docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+the last three hours:
+
+```
+@my_data.h2o_temperature{location, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["location"], window: 1h) 
+```
+
+Because the `aggregate` function is breaking the records out into one hour intervals, the `time` column is kept, 
+but the values are updated to the end of the window. For example, data at 2020-02-22T16:05 would get summarized in 
+a record with the `time` value (window) of `2020-02-22T17:00`. 
+
+| location | min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees | time |
+| --- | --- | --- | --- | --- |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T16:00 |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T17:00 |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T18:00 |
+| coyote_creek | 51 | 51 | 50 | 2020-02-22T17:00 |
+| coyote_creek | 50 | 50 | 50 | 2020-02-22T18:00 |
+| puget_sound | 40 | 40 | 40 | 2020-02-22T18:00 |
+
+Remember there were three series, so for 1h summaries for the last 3 hours we'd expect to have three rows per group 
+for nine total rows. But looking at the results, we only see the full three summaries for the `santa_monica` location. 
+This is because the other two series don't have data in every window of time.
+
+You can create rows for the missing summaries by calling `interpolate` after the call to aggregate. It can be used to 
+fill in default values or values based on a summary statistic of the rows like `first`, `min`, `max`, `mean`, etc. You 
+can find the [details for interpolate in the function reference]().
+
+#### Select
+The `select` function works similar to `aggregate` in that it operates on windows of time and groupings of rows.
+However, The difference is that select returns specific rows, rather than a summary data point. Here are a few 
+examples:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, surface_degrees, time > -3h}
+    |> select(fn: min("bottom_degrees"), by: [])
+```
+
+| location | state | bottom_degrees | surface_degrees | time
+| --- | --- | --- | --- | --- |
+| puget_sound | WA | 40 | 55 | 2020-02-22T17:10 |
+| puget_sound | WA | 40 | 54 | 2020-02-22T17:40 |
+
+Notice that all the columns are pulled through, the selected column name remains the same and note that 
+we've pulled in two records. We pull in two records because both of those have the same `min` `bottom_degrees` 
+value of `40`, which is the min for all records in that time. Notice also that `time` is the same value as those 
+individual records, which you don't see in calls to `aggregate`.
+
+You can also select in groups based on time or by other columns, just like you can with `aggregate`. Here's 
+an example using time grouping and inserting the time window for associated row selections:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> selector(fn: max(bottom_degrees), by: ["state"], window: 1h, windowColumn: "window")
```

Comment:
should this be `select`?

Reply:
If location isn't in my `by:` here, why is it returned? I thought fields that weren't in the `by:` got dropped?